src/yb/util/ulimit_info.cc

```diff
@@ -0,0 +1,89 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+// The following only applies to changes made to this file as part of YugaByte development.
+//
+// Portions Copyright (c) YugaByte, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.  See the License for the specific language governing permissions and limitations
+// under the License.
+//
+
+#include "yb/util/ulimit_info.h"
+
+#include <string>
+#include <sstream>
+#include <sys/time.h>
+#include <sys/resource.h>
+
+#include "yb/gutil/strings/substitute.h"
+#include "yb/util/env_util.h"
+#include "yb/util/path_util.h"
+
+using std::string;
+using std::stringstream;
+
+namespace yb {
+
+static stringstream& getLimit(
```

Comment:
The signature of this method should be more generic - something like

`Status getULimit(int resource, int64_t* soft_limit, int64_t* hard_limit)` and we can do the string conversion outside

Reply:
Done

Reply:
Done. Please review